components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/PolicyDAOImpl.java

```diff
@@ -90,71 +111,96 @@ public Policy getPolicy(String policyLevel, String policyName) throws APIMgtDAOE
     /**
      *@see PolicyDAO#addPolicy(String, Policy)
      */
+    @Override public void addPolicy(String policyLevel, Policy policy) throws APIMgtDAOException {
+
+        Connection connection;
+        try {
+            connection = DAOUtil.getConnection();
```

Comment:
Connection has not been closed. Please use ' try with resources' to ensure that DB resources such as PreparedStatements, ResultSets and Connections are closed automatically